packages/connection/src/Manager.php

```diff
@@ -1370,10 +1370,99 @@ public function verify_secrets( $action, $secret_1, $user_id ) {
 
 	/**
 	 * Responds to a WordPress.com call to authorize the current user.
-	 * Should be changed to protected.
+	 *
+	 * @param array $request The user authorization request data.
+	 * @return array|\IXR_Error Returns an array containing 'result'=>'authorized' on success.
+	 *                          Returns an IXR_Error on failure.
 	 */
-	public function handle_authorization() {
+	public function handle_authorization( $request ) {
+		$user = isset( $request['state'] ) ? get_user_by( 'id', $request['state'] ) : null;
+
+		/**
+		 * We've begun the remote authorization of the user.
+		 *
+		 * @since 8.0.0
+		 *
+		 * @param \WP_User $user The user object.
+		 */
+		do_action( 'jetpack_remote_authorize_begin', $user );
+
+		$return_error = function( \WP_Error $error ) use ( $user ) {
+			/**
+			 * User authorization has failed.
+			 *
+			 * @since 8.0.0
+			 *
+			 * @param \WP_User  $user The user object.
+			 * @param \WP_Error $error The error object.
+			 */
+			do_action( 'jetpack_remote_authorize_fail', $user, $error );
+
+			return $this->convert_to_ixr_error( $error );
+		};
+
+		foreach ( array( 'secret', 'state', 'redirect_uri', 'code' ) as $required ) {
+			if ( ! isset( $request[ $required ] ) || empty( $request[ $required ] ) ) {
+				return $return_error( new \WP_Error( 'missing_parameter', 'One or more parameters is missing from the request.', 400 ) );
+			}
+		}
+
+		if ( ! $user ) {
+			return $return_error( new \WP_Error( 'user_unknown', 'User not found.', 404 ) );
+		}
+
+		if ( $this->is_active() && $this->is_user_connected( $request['state'] ) ) {
+			return $return_error( new \WP_Error( 'already_connected', 'User already connected.', 400 ) );
+		}
 
+		$verified = $this->verify_secrets( 'authorize', $request['secret'], $request['state'] );
```

Comment:
Note that the previous `remote_authorize` method called `verify_action`. Since `verify_action`calls `verify_secrets`, let's just call `verify_secrets` directly.